File: app/src/main/java/com/github/kirillf/jobschedulerdemo/ServiceJob.java
package com.github.kirillf.jobschedulerdemo;

import android.app.job.JobParameters;
import android.app.job.JobService;

/**
 * Created by kirillf on 10/11/16.
 */

public class ServiceJob extends JobService {
    private static final String MESSAGE = "Started job with id %s";

    @Override
    public boolean onStartJob(JobParameters params) {
        ScheduledService.startScheduledJob(getApplicationContext(),
                String.format(MESSAGE, params.getJobId()));
        return false;
    }

    @Override
    public boolean onStopJob(JobParameters params) {
        return false;
    }
}

File: app/src/main/java/com/github/kirillf/jobschedulerdemo/ServiceJob.kt
package com.github.kirillf.jobschedulerdemo

import android.app.job.JobParameters
import android.app.job.JobService

/**
 * Created by kirillf on 10/11/16.
 */

class ServiceJob : JobService() {

    override fun onStartJob(params: JobParameters): Boolean {
        ScheduledService.startScheduledJob(applicationContext,
                String.format(MESSAGE, params.jobId))
        return false
    }

    override fun onStopJob(params: JobParameters): Boolean {
        return false
    }

    companion object {
        private const val MESSAGE = "Started job with id %s"
    }
}
